core/src/main/java/com/sequenceiq/cloudbreak/service/recipe/RecipeUsageService.java

```diff
@@ -0,0 +1,97 @@
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.common.usage.UsageProto;
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.domain.Recipe;
+import com.sequenceiq.cloudbreak.usage.UsageReporter;
+
+@Service
+public class RecipeUsageService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecipeUsageService.class);
+
+    @Inject
+    private UsageReporter usageReporter;
+
+    public void sendDetachedUsage(String stackCrn, Recipe recipe, String hostGroupName) {
+        try {
+            usageReporter.cdpRecipeEvent(getRecipeEventBuilder(stackCrn, recipe, hostGroupName)
+                    .setStatus(UsageProto.CDPRecipeStatus.Value.DETACHED)
+                    .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe detachment, because: ", e);
+        }
+    }
+
+    public void sendAttachedUsage(String stackCrn, Recipe recipe, String hostGroupName) {
+        try {
+            usageReporter.cdpRecipeEvent(getRecipeEventBuilder(stackCrn, recipe, hostGroupName)
+                    .setStatus(UsageProto.CDPRecipeStatus.Value.ATTACHED)
+                    .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe attachment, because: ", e);
+        }
+    }
+
+    public void sendDeletedUsage(Recipe recipe) {
+        try {
+            usageReporter.cdpRecipeEvent(getRecipeEventBuilder(recipe)
+                    .setStatus(UsageProto.CDPRecipeStatus.Value.DELETED)
+                    .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe deletion, because: ", e);
+        }
+    }
+
+    public void sendCreatedUsage(Recipe recipe) {
+        try {
+            usageReporter.cdpRecipeEvent(getRecipeEventBuilder(recipe)
+                    .setStatus(UsageProto.CDPRecipeStatus.Value.CREATED)
+                    .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe creation, because: ", e);
+        }
+    }
+
+    public void sendClusterCreationRecipeUsage(String stackCrn, int recipeCount, String typeDetails, String hostGroupDetails) {
+        try {
+            usageReporter.cdpClusterCreationRecipeEvent(UsageProto.CDPClusterCreationRecipeEvent.newBuilder()
+                            .setStackCrn(stackCrn)
+                            .setRecipeCount(recipeCount)
+                            .setHostGroupDetails(hostGroupDetails)
+                            .setTypeDetails(typeDetails)
+                            .build());
+        } catch (Exception e) {
+            LOGGER.error("Couldn't send usage report about recipe creation, because: ", e);
```

Comment:
Error message should not be the same as the create statement.
"Couldn't send usage report about cluster creation with recipe details, because:"